File: 240502/순위 경쟁2/ranking-competition2.py
N = int(input())
changes = []
for _ in range(N):
    c, s = input().split()
    ch_score = int(s)
    changes.append([c, ch_score])

a_score, b_score = 0, 0
change_cnts = 0
prev_winner, winner = '', ''
for ch_idx, (player, ch_score) in enumerate(changes):
    if player == 'A':
        a_score += ch_score
    elif player == 'B':
        b_score += ch_score

    if ch_idx != 0:
        if player == 'A':
            a_score += ch_score
        elif player == 'B':
            b_score += ch_score

        if a_score > b_score:
            winner = 'A'
        elif a_score < b_score:
            winner = 'B'
        elif a_score > 0 and a_score == b_score:
            change_cnts += 1
        elif a_score == 0 and a_score == b_score:
            continue

        if prev_winner != winner:
            change_cnts += 1
            prev_winner, winner = winner, ''
    else:
        if player == 'A':
            a_score += ch_score
            prev_winner, winner = player, player
        elif player == 'B':
            b_score += ch_score
            prev_winner, winner = player, player

        if a_score != 0 or b_score != 0:
            change_cnts += 1

print(change_cnts)

File: 260423/순위 경쟁2/ranking-competition2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    // Please write your code here.
    int N;
    cin >> N;

    vector<pair<char, int>> changes;
    for (int i = 0; i < N; i++) 
    {
        char c;
        int s;
        cin >> c >> s;
        changes.push_back({c, s});
    }

    int a_score = 0, b_score = 0;
    int change_cnts = 0;
    string prev_winner = "AB", winner = "AB";

    for (int ch_idx = 0; ch_idx < N; ch_idx++) 
    {
        char player = changes[ch_idx].first;
        int ch_score = changes[ch_idx].second;

        if (player == 'A') 
        {
            a_score += ch_score;
        } 
        else if (player == 'B') 
        {
            b_score += ch_score;
        }

        if (ch_idx != 0) 
        {
            if (a_score > b_score) 
            {
                winner = "A";
            } 
            else if (a_score < b_score) 
            {
                winner = "B";
            } 
            else 
            {
                winner = "AB";
            }
        } 
        else 
        {
            if (a_score < b_score) 
            {
                winner = "B";
            } 
            else if (a_score > b_score) 
            {
                winner = "A";
            }
        }

        if (prev_winner != winner) 
        {
            change_cnts++;
            prev_winner = winner;
            winner = "";
        }
    }

    cout << change_cnts << '\n';
    return 0;
}
